docs/application/dotnet/guides/nui/visuals.md

```diff
@@ -1,390 +1,552 @@
 
-<a name="bordervisual"></a>
-## Border Visual
+**Table: BorderVisual properties**
 
-The border visual renders a solid color as an internal border to the control's quad.
+| Property       | Type      | Required | Description                                                                          |
+|----------------|-----------|----------|--------------------------------------------------------------------------------------|
+| `Color`        | `Color`   | Yes      |  The color of the border.                                                            |
+| `BorderSize`   | `float`   | Yes      |  The width of the border in pixels.                                                  |
+| `AntiAliasing` | `bool`    | No       |  Specifies whether antialiasing of the border is required (default value - `false`). |
 
-**Figure: Border visual**
 
-![Border visual](media/border-visual.png)
 
-The following table lists the supported properties. The visual type is `Border` and the visual map is `BorderVisual`.
+**Usage:**
+The following example illustrates how to use a BorderVisual:
 
-**Table: Border visual properties**
+   ```
+   BorderVisual _borderVisual = new BorderVisual();
 
-| Property | Type | Required | Description |
-|----------|------|----------|-------------|
-| `Color` | `Vector4` | Yes |  The color of the border. |
-| `Size` |  `float`  | Yes |   The width of the border in pixels. |
-| `AntiAliasing` | `boolean` | No  |   Whether antialiasing of the border is required. |
+   // obligatory properties
+   _borderVisual.Color = Color.Red;
+   _borderVisual.BorderSize = 5.0f;
 
-**Usage:**
+   // optional properties:
+   // the size of the border will be interpreted as relative
+   _borderVisual.SizePolicy = VisualTransformPolicyType.Relative;
+   // the visual size (width and height) are equal half of the parent size             
+   _borderVisual.RelativeSize = new RelativeVector2(0.5f, 0.5f);
+   // position interpreted in absolute units
+   _borderVisual.PositionPolicy = VisualTransformPolicyType.Absolute;
+   // the reference point of the Visual is shifted right (50 pixels) and up (100 pixels) from the parent's reference point
+   _borderVisual.Position = new Vector2(50.0f, -100.0f);
+   // the parent's reference point
+   _borderVisual.Origin = Visual.AlignType.BottomBegin;
+   // point on the parent with respect to which the Visual will be placed
+   _borderVisual.AnchorPoint = Visual.AlignType.BottomBegin;
 
-The following example illustrates how to use a `BorderVisual` visual map. The visual is created with the `AddVisual()` method.
+   // _visualView is a previously created VisualView
+   _visualView.AddVisual("nameOfBorderVisual", _borderVisual);
+   ```
 
-```
-private BorderVisual borderVisualMap1;
 
-borderVisualMap1 = new BorderVisual();
 
-borderVisualMap1.Color = Color.Red;
-borderVisualMap1.BorderSize = 5.0f;
+**Figure: BorderVisual obtained from the above code (_visualView background color set to gray).**
```

Comment:
```suggestion 
 **Figure: BorderVisual after implementing example.**

Reply:
can we add the parentheses back?
in the examples I do not create VisualView, so in some of them I specify its color - this way you can see how the parent's and visualss reference points worked

Reply:
outdated